routing/control_tower.go

```diff
@@ -107,8 +128,18 @@ func (p *controlTower) InitPayment(paymentHash lntypes.Hash,
 func (p *controlTower) RegisterAttempt(paymentHash lntypes.Hash,
 	attempt *channeldb.HTLCAttemptInfo) error {
 
-	_, err := p.db.RegisterAttempt(paymentHash, attempt)
-	return err
+	p.paymentsMtx.Lock(paymentHash)
```

Comment:
Nice solution 👍

Reply:
I knew you'd like it, as the original inventor of multi-mutexing :)